src/main/java/com/google/cloud/tools/app/action/StageAction.java

```diff
@@ -26,71 +26,58 @@
 /**
  * Stages an application to be deployed.
  */
-public class StageAction extends AppAction {
+public class StageAction implements Action {
 
   private StageConfiguration configuration;
+  private StageExecutor stageExecutor;
 
-  public StageAction(StageConfiguration configuration) {
+  public StageAction(StageConfiguration configuration, StageExecutor stageExecutor) {
     Preconditions.checkNotNull(configuration);
     Preconditions.checkNotNull(configuration.getSourceDirectory());
     Preconditions.checkNotNull(configuration.getStagingDirectory());
-    Preconditions.checkNotNull(configuration.getAppEngineSdkRoot());
+    Preconditions.checkNotNull(stageExecutor);
 
     this.configuration = configuration;
+    this.stageExecutor = stageExecutor;
   }
 
   @Override
-  public boolean execute() {
+  public int execute() {
     List<String> arguments = new ArrayList<>();
-    arguments.add("stage");
     arguments.add(configuration.getSourceDirectory().toString());
     arguments.add(configuration.getStagingDirectory().toString());
-    if (configuration.isEnableQuickstart()) {
+    if (Boolean.TRUE.equals(configuration.isEnableQuickstart())) {
       arguments.add("--enable_quickstart");
     }
-    if (configuration.isDisableUpdateCheck()) {
+    if (Boolean.TRUE.equals(configuration.isDisableUpdateCheck())) {
       arguments.add("--disable_update_check");
     }
     if (!Strings.isNullOrEmpty(configuration.getVersion())) {
       arguments.add("--version");
       arguments.add(configuration.getVersion());
     }
-    if (!Strings.isNullOrEmpty(configuration.getCloudProject())) {
+    if (!Strings.isNullOrEmpty(configuration.getApplicationId())) {
```

Comment:
yeah, but it's still meaningful in the context of appengine AND stage still uses it, presumably we can change this after.